Build.props

```diff
@@ -0,0 +1,29 @@
+<Project>
+  <PropertyGroup>
+    <CodeAnalysisRuleSet>$(SolutionDir)Lombiq.Analyzers\General.ruleset</CodeAnalysisRuleSet>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.CodeAnalysis.FxCopAnalyzers" Version="3.0.0">
+      <PrivateAssets>all</PrivateAssets>
+      <IncludeAssets>runtime; build; native; contentfiles; analyzers</IncludeAssets>
+    </PackageReference>
+    <PackageReference Include="SonarAnalyzer.CSharp" Version="8.9.0.19135">
+      <PrivateAssets>all</PrivateAssets>
+      <IncludeAssets>runtime; build; native; contentfiles; analyzers</IncludeAssets>
+    </PackageReference>
+    <PackageReference Include="StyleCop.Analyzers" Version="1.1.118">
+      <PrivateAssets>all</PrivateAssets>
+    </PackageReference>
+  </ItemGroup>
+  
+  <Target Name="DisableAnalyzers" BeforeTargets="CoreCompile" Condition="'$(RunCodeAnalysis)' == 'false'"> 
+    <ItemGroup> 
+      <Analyzer Remove="@(Analyzer)" /> 
+    </ItemGroup> 
+  </Target>
+  
+  <Target Name="Copy editorconfig" AfterTargets="AfterResolveReferences">
+    <Copy SourceFiles="$(SolutionDir)Lombiq.Analyzers\.editorconfig" DestinationFolder="$(SolutionDir)" SkipUnchangedFiles="true" />
```

Comment:
When a project is built alone, without a solution, then `$SolutionDir` here will be undefined. We need to build projects individually sometimes in our CI, e.g. like this:

`dotnet build src\Utilities\Lombiq.Gulp.Extensions\Lombiq.Gulp.Extensions.csproj`

However, this failed with:

> Lombiq.Analyzers\Build.props(27,5): error MSB3030: Could not copy the file "*Undefined*Lombiq.Analyzers\.editorconfig" because it was not found.

Can we fix this? Or should we disable the analyzers in this case (something that if I'm not mistaken will work on later)?

Reply:
I've updated `Build.props` to use `$(MSBuildThisFileDirectory)` instead of `$(SolutionDir)`. It should fix a problem. Let me know if it doesn't.

Reply:
Looks good.